docarray/base_document/document.py

```diff
@@ -97,3 +99,24 @@ def __setattr__(self, field, value) -> None:
             for key, val in self.__dict__.items():
                 dict_ref[key] = val
             object.__setattr__(self, '__dict__', dict_ref)
+
+    def __eq__(self, cls) -> bool:
+        for key1,key2 in zip(self.dict(),cls.dict()):
+            if key1 != key2 :
+                return False
+            if key1 == "id" and key2 == "id":
+                continue
+            
+            value1 = self.dict()[key1]
+            value2 = cls.dict()[key2]
+
+            if isinstance(value1, np.ndarray) and isinstance(value2, np.ndarray):
+                if not np.array_equal(value1, value2):
```

Comment:
this will not cover TorchTensor and tensorflow. You need to rely on the abstracTensor class and the Computational backend. Probably you need to add `array_equal` in the computational backend

Reply:
I don't understand what you mean by adding `array_equal` in the computational backend.
Do you mean that I need to add `__eq__` and `array_equal` func to all of the Computational backend (`torch_backend` , `AbstractComputationalBackend` , etc)?

Reply:
Hey @samsja about relying on the Computational Backend and the abstractTensor class, do you mean by adding the `__eq__` in the backend file? 
If adding `__eq__` in the `abstract_computational_backend` class, is it enough by just adding the code below? If it is not, what should I compare in the backend and what should I add?
```
def __eq__(self : TTensor, other : TTensor) -> bool :
        if self.shape(TTensor) != other.shape(TTensor) :
            return False
```

Reply:
yes exactly adding this the to backend !

Shape checking is not enough. probably this should look like this

```python
def __eq__(self : TTensor, other : TTensor) -> bool :
        return (self == other).all()
```

Reply:
Thanks for the reply! Didn't know that actually works!